### ISSUE TITLE
Bug in writers when using Params to index Vars

### ISSUE THREAD

**Issue by blnicho** [Pyomo/pyomo#750]
The following toy model throws a key error:

```
from pyomo.environ import *
m = ConcreteModel()

m.p = Param(initialize=0)

m.s =  Set(initialize=[0,10])
m.x1 = Var(m.s, bounds=(0,10))

m.con = Constraint(expr=m.x1[0]<=10)
m.obj = Objective(expr=m.x1[m.p])

SolverFactory('glpk').solve(m,tee=True)
```
Error message:

```
Traceback (most recent call last):
  File "temp.py", line 12, in <module>
    SolverFactory('glpk').solve(m,tee=True)
  File "/home/blnicho/Research/pyomo/pyomo/opt/base/solvers.py", line 569, in solve
    self._presolve(*args, **kwds)
  File "/home/blnicho/Research/pyomo/pyomo/opt/solver/shellcmd.py", line 200, in _presolve
    OptSolver._presolve(self, *args, **kwds)
  File "/home/blnicho/Research/pyomo/pyomo/opt/base/solvers.py", line 669, in _presolve
    **kwds)
  File "/home/blnicho/Research/pyomo/pyomo/opt/base/solvers.py", line 740, in _convert_problem
    **kwds)
  File "/home/blnicho/Research/pyomo/pyomo/opt/base/convert.py", line 105, in convert_problem
    problem_files, symbol_map = converter.apply(*tmp, **tmpkw)
  File "/home/blnicho/Research/pyomo/pyomo/solvers/plugins/converter/model.py", line 88, in apply
    io_options=io_options)
  File "/home/blnicho/Research/pyomo/pyomo/core/base/block.py", line 1711, in write
    io_options)
  File "/home/blnicho/Research/pyomo/pyomo/repn/plugins/cpxlp.py", line 175, in __call__
    include_all_variable_bounds=include_all_variable_bounds)
  File "/home/blnicho/Research/pyomo/pyomo/repn/plugins/cpxlp.py", line 697, in _print_model_LP
    column_order)
  File "/home/blnicho/Research/pyomo/pyomo/repn/plugins/cpxlp.py", line 217, in _print_expr_canonical
    names = [variable_symbol_dictionary[id(var)] for var in x.linear_vars]
  File "/home/blnicho/Research/pyomo/pyomo/repn/plugins/cpxlp.py", line 217, in <listcomp>
    names = [variable_symbol_dictionary[id(var)] for var in x.linear_vars]
KeyError: 139870770319848
```

The same error is thrown in the ampl writer if the solver in the toy model is changed to Ipopt.

**Comment by Homersmyid** [comment]
Second to last line should be:
m.obj = Objective(expr=m.x1[value(m.p)])

**Comment by jsiirola** [comment]
This was resolved by #1006.
